### PR TITLE
Temporal 102 Code Reorganization (Go)

### PR DESCRIPTION
**Pull request by angelazhou32**
This PR reorganizes the code repository for Temporal 102 in Go such as moving the structs exercise to a sample instead of an exercise, remove versioning etc.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,17 +9,16 @@ training course.
...
+`exercises/testing-code`           | [Exercise 2](exercises/testing-code/README.md)
+`exercises/debug-activity`         | [Exercise 3](exercises/debug-activity/README.md)
 
 
 ## Examples for Self-Study
 Directory Name                         | Description
 :------------------------------------- | :----------------------------------------------------------------------------------
-`samples/age-estimation`               | [Calls a remote API to estimate a given name](samples/age-estimation)
+`samples/age-estimation`               | [Calls a remote API to estimate age of a person given their name](samples/age-estimation)
+`samples/using-structs-for-data`               | [Demonstrates how structs are passed in as input and output data](samples/using-structs-for-data)
```
Comment (axfelix) [inline README.md]: table broken? otherwise this looks 100%

Comment (angelazhou32) [reply README.md]: Thanks :) Fixed!

**Review by axfelix — APPROVED** [review]

**Review by MasonEgger — APPROVED** [review]
LGTM

File: README.md [hunk 2]
```diff
@@ -9,17 +9,16 @@ training course.
...
+`exercises/durable-execution`      | [Exercise 1](exercises/durable-execution/README.md)
+`exercises/testing-code`           | [Exercise 2](exercises/testing-code/README.md)
+`exercises/debug-activity`         | [Exercise 3](exercises/debug-activity/README.md)
 
 
 ## Examples for Self-Study
 Directory Name                         | Description
 :------------------------------------- | :----------------------------------------------------------------------------------
-`samples/age-estimation`               | [Calls a remote API to estimate a given name](samples/age-estimation)
+`samples/age-estimation`               | [Calls a remote API to estimate age of a person given their name](samples/age-estimation)
```
Comment (tomwheeler) [inline README.md:20]: Nice suggestion on the wording change—I didn't notice until now that "given name" could have two interpretations here.

**Review by tomwheeler — APPROVED** [review]
Perfect. I tested the sample in its new location and it worked as expected.

**Pull request merged**